src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -224,23 +224,23 @@ Value MathBuilder::negativeInf(Type type) const {
           [&](Type) { value = -std::numeric_limits<double>::infinity(); })
       .Case<IntegerType>([&](IntegerType type) {
         unsigned width = type.getWidth();
-        bool isSigned = type.isSigned();
+        bool isSignless = type.isSignlessInteger();
         switch (width) {
         case 8:
-          value = (isSigned) ? std::numeric_limits<int8_t>::min()
-                             : std::numeric_limits<uint8_t>::min();
+          value = (isSignless) ? std::numeric_limits<int8_t>::min()
```

Comment:
I agree that a signless integer min value is conservatively `std::numeric_limits<int8_t>::min()`, however if the integer is not signless it might be signer or unsigned. If it is signed the min value should also be `std::numeric_limits<int8_t>::min()`, right ?

Suggests:

```
 value = (isSignless || isSigned) ? std::numeric_limits<int8_t>::min()
                                                      : std::numeric_limits<uint8_t>::min();

Reply:
I believe MLIR treat signless as signed...

Reply:
Sure, that is fine. My suggestion is to handle singed types. If the type is a signed integer the code change I suggested will use the min value for ont8_t rather than the min value for uint8_t.

Reply:
[MLIR constant](https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/Dialect/Arithmetic/IR/ArithmeticOps.td#L120) only supports signless at this moment. Anyway, I added `(isSignless || isSigned)` just in case signed constant will be supported in the future, but it would fail if users use signed integers at this moment.

Reply:
This message shows MLIR constant only supports signless.
```
within split at test-reduce.mlir:90 offset :4:10: error: 'arith.constant' op integer return type must be signless
  %cst = arith.constant 1: si64
```